learning_observer/learning_observer/canvas.py

```diff
@@ -0,0 +1,341 @@
+import os
+import json
+import string
+import requests
+import recordclass
+import configparser
+import aiohttp
+import aiohttp.web
+import pmss
+import yaml
+
+import learning_observer.settings as settings
+import learning_observer.log_event
+import learning_observer.util
+import learning_observer.auth
+import learning_observer.runtime
+
+cache = None
+
+pmss.register_field(
+    name="default_server",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The default server for Canva",
+    required=True
+)
+pmss.register_field(
+    name="client_id",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The client ID for Canva",
+    required=True
+)
+pmss.register_field(
+    name="client_secret",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The client secret for Canva",
+    required=True
+)
+pmss.register_field(
+    name="access_token",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The access token for Canva",
+    required=True
+)
+pmss.register_field(
+    name="refresh_token",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The refresh token for Canva",
+    required=True
+)
```

Comment:
This was used to access the Canvas auth details using the PMSS style from the creds.yaml but I mentioned that I need to update the access_token programmatically and it did not feel safe to do to the creds.yaml file because a lot depended on it. 

So, should I create a new yaml file and access using PMSS or config.ini is fine as used?

Reply:
Usually the authorization tokens live within the request object that's passed around. 

The general workflow is, user signs-in with Google and their authorization token is stored with their session/request. When requests are made to the google api, the token is added to the api call.

It should be similar for canvas and schoology.

Reply:
I've implemented the authorization token for Canvas too (and it does not require a config.ini file anymore)